app/services/teacher_training_public_api/sync_courses.rb

```diff
@@ -43,11 +43,27 @@ def create_or_update_course(course_from_api)
         assign_course_attributes(course, course_from_api, recruitment_cycle_year)
         add_accredited_provider(course, course_from_api[:accredited_body_code], recruitment_cycle_year)
 
+        notify_candidates = visa_deadline_has_changed(course)
+
         course.save!
+        if notify_candidates == true
+          CandidateMailers::EnqueueVisaSponsorshipDeadlineChangeWorker.perform_async(course.id)
+        end
+
         course
       end
     end
 
+    def visa_deadline_has_changed(course)
```

Comment:
I think we need to return false if the course goes from having a deadline to not having a deadline.